src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -83,39 +86,46 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
     }
 
-    // Helper method to create term queries with variations for spaces
-    private BoolQueryBuilder createNameMatchQuery(String field, String name) {
-        return QueryBuilders.boolQuery()
-            .should(QueryBuilders.termQuery(field, name)) // Exact match
-            .should(QueryBuilders.termQuery(field, " " + name)) // Prepend space
-            .should(QueryBuilders.termQuery(field, name + " ")) // Append space
-            .should(QueryBuilders.termQuery(field, " " + name + " ")); // Prepend and append space
+    public String[] normalizeAndSplit(String name) {
+        // remove "name changed by Deed Poll" ignoring case
+        String nameWithoutDeedPollStatement = name.replaceAll("(?i)name changed by deed poll", "").trim();
+
+        // split to separate names for search wherever there are non alphanumeric characters
+        return Arrays.stream(nameWithoutDeedPollStatement.split("[^a-zA-Z0-9]+"))
```

Comment:
I think this will remove non-ASCII characters which might be a valid part of the name, e.g. European names with accents ("é"), Polish names with ogoneks ("ę"), Chinese symbols and others. We could put in a weaker regex that only excludes punctuation or accept that some letters won't be matched if we think the impact will be low

Reply:
I've updated the code to cope not on the unicode and added to tests